kolibri/core/tasks/api.py

```diff
@@ -307,1225 +130,110 @@ def create(self, request):
+        )
         return Response(job_response)
 
-    def destroy(self, request, pk=None):
-        # unimplemented for now.
-        pass
-
-    @decorators.action(methods=["post"], detail=False)
-    def canceltask(self, request):
-        """
-        Cancel a task with its task id given in the task_id parameter.
-        """
-
-        if "task_id" not in request.data:
-            raise serializers.ValidationError("The 'task_id' field is required.")
-        if not isinstance(request.data["task_id"], string_types):
-            raise serializers.ValidationError("The 'task_id' should be a string.")
-
-        for _queue in self.queues:
-            try:
-                _queue.cancel(request.data["task_id"])
-                break
-            except JobNotFound:
-                continue
-
-        return Response({})
-
-    @decorators.action(methods=["post"], detail=False)
-    def cleartasks(self, request):
-        """
-        Cancels all running tasks.
-        """
-
-        for _queue in self.queues:
-            _queue.empty()
-
-        return Response({})
-
-    @decorators.action(methods=["post"], detail=False)
-    def cleartask(self, request):
-        # Given a single task ID, clear it from the queue
-        task_id = request.data.get("task_id")
-        if not task_id:
-            return Response({})
-
-        for _queue in self.queues:
-            _queue.clear_job(task_id)
-
-        return Response({"task_id": task_id})
-
-    @decorators.action(methods=["post"], detail=False)
-    def deletefinishedtasks(self, request):
+    @decorators.action(methods=["post"], detail=True)
+    def cancel(self, request, pk=None):
         """
-        Delete all tasks that have succeeded, failed, or been cancelled.
+        Cancel a task with its job id given in the url parameter.
         """
-        task_id = request.data.get("task_id")
-        if task_id:
-            for _queue in self.queues:
-                _queue.clear_job(task_id)
-        else:
-            for _queue in self.queues:
-                _queue.clear()
-        return Response({})
-
-
-class TasksViewSet(BaseViewSet):
-    @property
-    def queues(self):
-        return [queue, priority_queue]
-
-    def default_permission_classes(self):
-        # Permission for /api/tasks/tasks/ endpoints.
-        if self.action in ["list", "retrieve", "create", "restarttask"]:
-            return [IsAuthenticated | NotProvisionedHasPermission]
-        elif self.action in ["deletefinishedtasks"]:
-            return [CanManageContent | CanExportLogs]
-        elif self.action == "startexportlogcsv":
-            return [CanExportLogs]
-        elif self.action in ["importusersfromcsv", "exportuserstocsv"]:
-            return [CanImportUsers]
-
-        # For all other task endpoints.
-        return [CanManageContent]
-
-    @decorators.action(methods=["post"], detail=False)
-    def startchannelupdate(self, request):
```

Comment:
all these taks are being implemented in upcoming PRs, aren't they?

Reply:
Yes, I have the code and full updates for all of the backend tasks.